bin/rialto/runtime/src/millau_messages.rs

```diff
@@ -16,21 +16,41 @@
 
 //! Everything required to serve Millau <-> Rialto message lanes.
 
-use bridge_runtime_common::messages;
+use crate::Runtime;
 
 use bp_message_lane::{
 	source_chain::TargetHeaderChain,
 	target_chain::{ProvedMessages, SourceHeaderChain},
-	InboundLaneData, LaneId, Message, MessageNonce,
+	InboundLaneData, LaneId, Message, MessageKey, MessageNonce,
 };
 use bp_runtime::InstanceId;
-use bridge_runtime_common::messages::MessageBridge;
+use bridge_runtime_common::messages::{self, MessageBridge};
 use frame_support::{
+	storage::generator::StorageMap,
 	weights::{Weight, WeightToFeePolynomial},
 	RuntimeDebug,
 };
+use pallet_message_lane::{DefaultInstance, InboundLanes, OutboundLanes, OutboundMessages};
+use sp_core::storage::StorageKey;
 use sp_trie::StorageProof;
 
+/// Storage key of the Rialto -> Millau message in the runtime storage.
+pub fn message_key(lane: &LaneId, nonce: MessageNonce) -> StorageKey {
+	let message_key = MessageKey { lane_id: *lane, nonce };
+	let raw_storage_key = OutboundMessages::<Runtime, DefaultInstance>::storage_map_final_key(message_key);
+	StorageKey(raw_storage_key)
+}
+
+/// Storage key of the Rialto -> Millau message lane state in the runtime storage.
+pub fn outbound_lane_data_key(lane: &LaneId) -> StorageKey {
+	StorageKey(OutboundLanes::<DefaultInstance>::storage_map_final_key(*lane))
+}
+
+/// Storage key of the Millau -> Rialto message lane state in the runtime storage.
+pub fn inbound_lane_data_key(lane: &LaneId) -> StorageKey {
+	StorageKey(InboundLanes::<Runtime>::storage_map_final_key(*lane))
```

Comment:
If someone could point me to the issue/typo with `OutboundLanes`/`InboundLanes`, I would be very grateful. As you can see, I have to use `::<DefaultInstance>::` with `OutboundLanes`, but `::<Runtime>::` with `InboundLanes`. All other options lead to compilation error. Both maps are almost identical, so I'm not sure - why's that? Also, I guess, it'll lead to incorrect storage keys returned => incorrect proofs. But I can't prove that right now.

Reply:
I think the difference has to do with the `OutboundLanes` map not requiring access to `Runtime`'s associated types (via `T`). This means it just needs something that satisfies the `Instance` type. On the other hand since `InboundLanes` uses `T::InboundRelayer` it requires `Runtime` since that where the associated type comes from. 

What you have now with`InboundLanes::<Runtime>::storage_map_final_key(*lane)` is the same as `<Runtime, DefaultInstance>`, so the types will work out. I'm not sure what problems we may run into if they don't though.

We might want to make these generic over `Instance` so we're not tied to only using the default instance of the pallet.

Reply:
I guess that's the issue, thanks!